include/evhtp/evhtp.h

```diff
@@ -1370,6 +1373,16 @@ EVHTP_EXPORT struct bufferevent * evhtp_connection_take_ownership(evhtp_connecti
 EVHTP_EXPORT void evhtp_connection_free(evhtp_connection_t * connection);
 EVHTP_EXPORT void evhtp_request_free(evhtp_request_t * request);
 
+/**
+ * @brief set a max headers size to accept for an incoming request, this will
+ *        default to unlimited.
+ *        note: the headers size includes request line.
```

Comment:
When I think "Header" in the context of HTTP, I think `header: value`, not 

```HTTP
GET / ...\r\n\r\n
Header: value\r\n\r\n
```

This could be confusing.

There are a few interesting hooks you can try:

In the parser we have:

```C
htparse_hook  on_hdrs_begin;
htparse_data_hook hdr_key;
htparse_data_hook hdr_val;
htparse_hook   on_hdrs_complete;
```

Which in libevhtp maps to:
```C
static htparse_hooks request_psets = {
    /* ..... removed cruft ...... */
    .on_hdrs_begin      = htp__request_parse_headers_start_,
    .hdr_key            = htp__request_parse_header_key_,
    .hdr_val            = htp__request_parse_header_val_,
    .on_hdrs_complete   = htp__request_parse_headers_,
    /* ...... */
};
```

Reply:
Because libevent's http is also like this, so I want to agree with it.

Reply:
```
HTTP-message   = Request | Response     ; HTTP/1.1 messages
        Request       = Request-Line              ; Section 5.1
                        *(( general-header        ; Section 4.5
                         | request-header         ; Section 5.3
                         | entity-header ) CRLF)  ; Section 7.1
                        CRLF
                        [ message-body ]          ; Section 4.3
```

Seems as if RFC designates the `GET|POST /.... HTTP/N.N` as the request, and `request-header|entity-header` as "headers". Otherwise they are all encapsulated inside a `HTTP-message`. So Libevent is wrong IMO.